package/kedro_viz/integrations/deployment/azure_deployer.py

```diff
@@ -0,0 +1,84 @@
+"""`kedro_viz.integrations.deployment.azure_deployer` defines
+deployment class for Azure Blob Storage"""
+import glob
+import logging
+import mimetypes
+from pathlib import Path
+
+from kedro_viz.integrations.deployment.base_deployer import BaseDeployer
+
+try:
+    from azure.storage.blob import ContentSettings
+except ImportError:  # pragma: no cover
+    pass
+
+import fsspec
+
+from kedro_viz import __version__
+
+_AZ_PROTOCOL = "abfs"
+logger = logging.getLogger(__name__)
+
+
+class AzureDeployer(BaseDeployer):
+    """A class to handle the deployment of Kedro-viz to AzureBlobStorage.
+
+    Attributes:
+        _endpoint (str): Azure endpoint of the hosted site.
+        _bucket_name (str): Name of the AzureBlobStorage account.
+        _path (str): Container path for the AzureBlobStorage account.
+        _fs (fsspec.filesystem): Filesystem for Azure protocol.
+    """
+
+    def __init__(self, endpoint, bucket_name):
+        """Initialize AzureBlobStorage with endpoint and bucket name.
+
+        Args:
+            endpoint (str): Azure endpoint of the hosted site.
+            bucket_name (str): Name of the AzureBlobStorage account.
+        """
+        super().__init__()
+        self._endpoint = endpoint
+        self._bucket_name = bucket_name
+        self._path = f"{_AZ_PROTOCOL}://$web"
```

Comment:
What is `$web`?

Reply:
$web is the default container in AzureBlobStorage when someone enables static website hosting. We will mention this in the docs that the users need to have this container to use static hosting.